weapon.py

```diff
@@ -2,23 +2,47 @@
 
 class Weapon:
 
-    def __init__(self, damage, name, rarity, cost):
-        self.damage = damage
+    def __init__(self, dice, num_rolls, name, rarity, cost):
+        self.dice = dice
+        self.num_rolls = num_rolls
         self.name = name
         self.rarity = rarity
         self.cost = cost
 
+    #@property
+    def damage(self):
+        return self.dice.roll(self.num_rolls)
+
     def __repr__(self):
-        return f"\n{self.name} deals damage {self.damage}, rarity: {self.rarity}, cost: {self.cost}"
+        return f"\n{self.name}, damage: {self.num_rolls}d{self.dice}, rarity: {self.rarity}, cost: {self.cost}"
 
 weapons = [
-    Weapon(Dice(1, 6).show_dice(), "Long sword", 0.5, 100),
-    Weapon(Dice(2, 6).show_dice(), "Wolf Knight's Greatsword", 0.8, 800),
-    Weapon(Dice(1, 12).show_dice(), "Uchigatana", 0.6, 500),
-    Weapon(Dice(2, 4).show_dice(), "Sellsword Twinblades", 0.5, 200),
-    Weapon(Dice(1, 4).show_dice(), "Dagger", 0.4, 50)
+    Weapon(Dice(6), 1, "Long sword", 0.5, 100),
+    Weapon(Dice(6), 2, "Wolf Knight's Greatsword", 0.8, 800),
+    Weapon(Dice(12), 1, "Uchigatana", 0.6, 500),
+    Weapon(Dice(4), 2, "Sellsword Twinblades", 0.5, 200),
+    Weapon(Dice(4), 1, "Dagger", 0.4, 50)
```

Comment:
Learn how to declare data in json and load weapon parameters from json-configuration file.